src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -157,6 +157,7 @@ internal static IPHostEntry InternalGetHostByAddress(IPAddress address, bool inc
         public static string GetHostName()
         {
             GlobalLog.Print("Dns.GetHostName");
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
The main reason for moving was the calls to Socket.OSSupportsIPv6 which used to force the initialization but now it expects it has been done already. 
I agree that it make sense for these to be in PAL: that way the *NIX PALs don't even need to make the extra (no-op) call. (Fixing.)